Add examples for CSR-COO storage format

taruns900 opened the pull request:
### What does this PR do?

This Pull Request updates the `CMakeLists.txt` file to correctly include and compile the newly added standalone `HybridCSR_COO` example files.

### Which issue does this PR close?

* Closes #21.

### Rationale for this change

This change is necessary to ensure that the previously added `HybridCSR_COO` standalone usage examples (`HybridCSRExample1.cpp`, `HybridCSRExample2.cpp`, and `HybridCSRExample3.cpp`) are properly integrated into the project's build system. Without this update, these new examples would not be compiled or linked, preventing users from running them.

### What changes are included in this PR?

This PR includes the following change:

* **Update to `CMakeLists.txt`**: Modified the `if(BUILD_EXAMPLES)` block to explicitly `list(APPEND ...)` the paths to the three new `HybridCSR_COO` example files (`examples/extras/HybridCSRExample1.cpp`, `examples/extras/HybridCSRExample2.cpp`, and `examples/extras/HybridCSRExample3.cpp`) to the `EXAMPLE_SOURCES` variable. This ensures they are processed by CMake and built into executables.

### Are these changes tested?

Yes, these changes are implicitly tested by the successful compilation and execution of the new `HybridCSR_COO` example programs themselves. The primary goal of this PR is to enable their build, and their successful build confirms the correctness of this `CMakeLists.txt` modification.

### Are there any user-facing changes?

No, this PR itself does not introduce direct user-facing changes to the application's functionality. It is a build system update that enables the compilation of new user-facing example programs.

taruns900 committed:
Add standalone usage examples for HybridCSR_COO

taruns900 committed:
feat: Include HybridCSR_COO standalone examples in CMake build

taruns900 committed:
Merge branch 'SharonIV0x86:main' into main

SharonIV0x86 commented on CMakeLists.txt:
```diff
@@ -40,6 +40,14 @@ endif()
 if(BUILD_EXAMPLES)
     file(GLOB_RECURSE EXAMPLE_SOURCES CONFIGURE_DEPENDS ${CMAKE_SOURCE_DIR}/examples/*.cpp)
+
+    # Add the new HybridCSR_COO examples
```
Instead of manually mentioning the file paths here use a recurse to iterate over the ``/examples/extras`` directory and compile any file ending with ``.cpp``.

SharonIV0x86 commented on examples/extras/HybridCSRExample1.cpp:
```diff
@@ -0,0 +1,68 @@
+    int src = 0;
+    int dest = 1;
+    std::cout << "Querying edge from " << src << " to " << dest << ": ";
+    // Assuming HybridCSR_COO has a method to get edge weight, e.g., getEdgeWeight or similar.
```
Try to demonstrate the use of ``impl`` function of the hybrid class in this example.

SharonIV0x86 commented on examples/extras/HybridCSRExample2.cpp:
```diff
@@ -0,0 +1,33 @@
+
+    HybridCSR_COO<std::string, double> hybridStorage;
+    hybridStorage.populateFromAdjList(adjList);
+
```
Try to demonstrate the use of ``impl`` functions of the hybrid class in this example as well.

SharonIV0x86 commented on examples/extras/HybridCSRExample3.cpp:
```diff
@@ -0,0 +1,38 @@
+
+    std::cout << "Graph populated with disconnected components and self-loops." << std::endl;
+    std::cout << "Successful population demonstrates handling of these graph structures." << std::endl;
+
```
Try to demonstrate the use of ``impl`` functions of the hybrid class in this example too.

SharonIV0x86 reviewed (changes requested):
Needs some changes

taruns900 committed:
Refactor CMake and use impl functions in HybridCSR_COO examples

taruns900 commented:
# Full Changes Details

## `CMakeLists.txt`

### Change:  
Modified the build configuration for examples to use a recursive `file(GLOB_RECURSE ...)` command to automatically discover all `.cpp` files in:  
- `examples/`  
- `examples/extras/`  

### Rationale:  
- **Automation**: Eliminates the need to manually list each new example file.  
- **Maintainability**: Future `.cpp` files added to these directories will be automatically included in the build.  
- **Scalability**: Simplifies project expansion as the number of examples grows.  

---

## Example File Updates

### `examples/extras/HybridCSRExample1.cpp`  
#### Change:  
Added post-`populateFromAdjList` code demonstrating:  
```cpp
hybridGraph.impl_getEdge(src, dest);

SharonIV0x86 commented on examples/extras/HybridCSRExample1.cpp:
```diff
@@ -0,0 +1,50 @@
+    std::cout << "--- HybridCSR_COO Example 1: Basic Population and Query ---" << std::endl;
+
+    // Create a dummy adjacency list (vertex -> vector of (neighbor, weight))
+    std::unordered_map<int, std::vector<std::pair<int, int>>> adjList = {
```
```suggestion
    std::unordered_map<int, std::vector<std::pair<int, int>>, CinderPeak::VertexHasher<int, void>> adjList = {
```

SharonIV0x86 commented on examples/extras/HybridCSRExample2.cpp:
```diff
@@ -0,0 +1,39 @@
+
+    // Dummy unweighted adjacency list (vertex -> vector of (neighbor, weight))
+    // We use a dummy weight (0) for an unweighted graph
+    std::unordered_map<std::string, std::vector<std::pair<std::string, int>>> adjList = {
```
```suggestion
    std::unordered_map<std::string, std::vector<std::pair<std::string, int>>, CinderPeak::VertexHasher<std::string, void>> adjList = {

```

SharonIV0x86 commented on examples/extras/HybridCSRExample3.cpp:
```diff
@@ -0,0 +1,55 @@
+
+using namespace CinderPeak::PeakStore;
+
+class CustomVertex {
```
```suggestion
class CustomVertex : public CinderPeak::CinderVertex{
```

SharonIV0x86 commented on examples/extras/HybridCSRExample3.cpp:
```diff
@@ -0,0 +1,55 @@
+};
+
+// Hash function for CustomVertex
+namespace std {
```
Remove this std namespace for hash function.

SharonIV0x86 commented on examples/extras/HybridCSRExample3.cpp:
```diff
@@ -0,0 +1,55 @@
+    std::cout << "--- HybridCSR_COO Example 3: Custom Vertex and Edge Types ---" << std::endl;
+
+    // Dummy adjacency list with custom vertices
+    std::unordered_map<CustomVertex, std::vector<std::pair<CustomVertex, int>>> adjList = {
```
```suggestion
    std::unordered_map<CustomVertex, std::vector<std::pair<CustomVertex, int>>, CinderPeak::VertexHasher<CustomVertex>> adjList = {

```

SharonIV0x86 reviewed (changes requested):
Need minor changes.

SharonIV0x86 commented:
@taruns900 Status of your PR?
